File: pkg/store/model/tool_config.go
package model

import (
	"time"
)

// ToolAIConfig 工具AI配置
type ToolAIConfig struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`

	// DeepSeek 配置
	DeepSeekEnabled bool   `gorm:"default:false" json:"deepseek_enabled"`
	DeepSeekAPIKey  string `gorm:"size:255" json:"deepseek_api_key,omitempty"`
	DeepSeekModel   string `gorm:"size:50;default:'deepseek-chat'" json:"deepseek_model"`

	// Gemini 配置
	GeminiEnabled   bool   `gorm:"default:false" json:"gemini_enabled"`
	GeminiAPIKey    string `gorm:"size:255" json:"gemini_api_key,omitempty"`
	GeminiAPIKeys   string `gorm:"type:text" json:"gemini_api_keys,omitempty"` // JSON数组
	GeminiModel     string `gorm:"size:50;default:'gemini-2.0-flash'" json:"gemini_model"`
	GeminiTimeout   int    `gorm:"default:120" json:"gemini_timeout"`
	GeminiMaxTokens int    `gorm:"default:8000" json:"gemini_max_tokens"`

	// OpenAI 兼容配置
	OpenAIEnabled   bool   `gorm:"default:false" json:"openai_enabled"`
	OpenAIProvider  string `gorm:"size:50" json:"openai_provider,omitempty"`
	OpenAIAPIKey    string `gorm:"size:255" json:"openai_api_key,omitempty"`
	OpenAIBaseURL   string `gorm:"size:255" json:"openai_base_url,omitempty"`
	OpenAIModel     string `gorm:"size:50;default:'gpt-3.5-turbo'" json:"openai_model"`
	OpenAITimeout   int    `gorm:"default:60" json:"openai_timeout"`
	OpenAIMaxTokens int    `gorm:"default:4000" json:"openai_max_tokens"`

	// Baidu 配置
	BaiduEnabled bool   `gorm:"default:false" json:"baidu_enabled"`
	BaiduAppID   string `gorm:"size:100" json:"baidu_app_id,omitempty"`
	BaiduSecret  string `gorm:"size:255" json:"baidu_secret,omitempty"`

	// 备注
	Notes string `gorm:"type:text" json:"notes,omitempty"`
}

// ToolPreference 工具全局偏好
type ToolPreference struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`

	// 任务默认设置
	DefaultAutoUpload    bool   `gorm:"default:true" json:"default_auto_upload"`
	DefaultUploadDelay   int    `gorm:"default:10" json:"default_upload_delay"`
	DefaultSubtitleDelay int    `gorm:"default:10" json:"default_subtitle_delay"`
	DefaultCopyright     int    `gorm:"default:2" json:"default_copyright"`
	DefaultSource        string `gorm:"default:'YouTube'" json:"default_source"`
	DefaultTid           int    `gorm:"default:122" json:"default_tid"`

	// 界面设置
	Theme        string `gorm:"default:'light';size:20" json:"theme"` // light/dark
	Language     string `gorm:"default:'zh';size:10" json:"language"` // zh/en
	ItemsPerPage int    `gorm:"default:20" json:"items_per_page"`
	ShowAdvanced bool   `gorm:"default:false" json:"show_advanced"`

	// 隐私设置
	EnableAnalytics bool `gorm:"default:false" json:"enable_analytics"`
}

// TableName 指定表名
func (ToolAIConfig) TableName() string {
	return "cw_user_ai_configs"
}

func (ToolPreference) TableName() string {
	return "cw_user_preferences"
}

File: web/src/lib/toolConfig.ts
/**
 * 全局工具配置 API 封装
 */

// 获取 API 基础 URL
const getApiBaseUrl = () => {
    if (typeof window !== 'undefined') {
        const { protocol, hostname, port } = window.location;
        return `${protocol}//${hostname}${port ? ':' + port : ''}`;
    }
    return 'http://localhost:8096';
};

const getRequestHeaders = (): HeadersInit => ({ 'Content-Type': 'application/json' });

// 工具 AI 配置类型
export interface ToolAIConfig {
    id?: number;
    // DeepSeek
    deepseek_enabled: boolean;
    deepseek_api_key: string;
    deepseek_model: string;
    // Gemini
    gemini_enabled: boolean;
    gemini_api_key: string;
    gemini_api_keys: string;
    gemini_model: string;
    gemini_timeout: number;
    gemini_max_tokens: number;
    // OpenAI Compatible
    openai_enabled: boolean;
    openai_api_key: string;
    openai_base_url: string;
    openai_model: string;
    openai_provider: string;
    openai_timeout: number;
    openai_max_tokens: number;
    // Baidu
    baidu_enabled: boolean;
    baidu_app_id: string;
    baidu_secret: string;
}

// 全局偏好类型
export interface ToolPreference {
    id?: number;
    // 上传默认值
    default_auto_upload: boolean;
    default_upload_delay: number;
    default_subtitle_delay: number;
    default_copyright: number;
    default_source: string;
    default_tid: number;
    // UI 偏好
    theme: string;
    language: string;
    items_per_page: number;
    show_advanced: boolean;
    enable_analytics: boolean;
}

// API 响应类型
interface ApiResponse<T> {
    code: number;
    message: string;
    data?: T;
}

/**
 * 工具配置 API
 */
export const toolConfigApi = {
    // ==================== AI 配置 ====================

    /**
     * 获取工具 AI 配置
     */
    async getAIConfig(): Promise<ToolAIConfig | null> {
        try {
            const res = await fetch(`${getApiBaseUrl()}/api/v1/tool/config/ai`, {
                headers: getRequestHeaders(),
            });
            const data: ApiResponse<ToolAIConfig> = await res.json();
            if (data.code === 0 && data.data) {
                return data.data;
            }
            return null;
        } catch (e) {
            console.error('获取工具 AI 配置失败:', e);
            return null;
        }
    },

    /**
     * 更新工具 AI 配置
     */
    async updateAIConfig(config: Partial<ToolAIConfig>): Promise<boolean> {
        try {
            const res = await fetch(`${getApiBaseUrl()}/api/v1/tool/config/ai`, {
                method: 'PUT',
                headers: getRequestHeaders(),
                body: JSON.stringify(config),
            });
            const data: ApiResponse<ToolAIConfig> = await res.json();
            return data.code === 0;
        } catch (e) {
            console.error('更新工具 AI 配置失败:', e);
            return false;
        }
    },

    // ==================== 偏好设置 ====================

    /**
     * 获取工具偏好
     */
    async getPreferences(): Promise<ToolPreference | null> {
        try {
            const res = await fetch(`${getApiBaseUrl()}/api/v1/tool/config/preferences`, {
                headers: getRequestHeaders(),
            });
            const data: ApiResponse<ToolPreference> = await res.json();
            if (data.code === 0 && data.data) {
                return data.data;
            }
            return null;
        } catch (e) {
            console.error('获取工具偏好失败:', e);
            return null;
        }
    },

    /**
     * 更新工具偏好
     */
    async updatePreferences(prefs: Partial<ToolPreference>): Promise<boolean> {
        try {
            const res = await fetch(`${getApiBaseUrl()}/api/v1/tool/config/preferences`, {
                method: 'PUT',
                headers: getRequestHeaders(),
                body: JSON.stringify(prefs),
            });
            const data: ApiResponse<ToolPreference> = await res.json();
            return data.code === 0;
        } catch (e) {
            console.error('更新工具偏好失败:', e);
            return false;
        }
    },
};

export default toolConfigApi;
